docs/rfc/rfc-5.adoc

```diff
@@ -0,0 +1,121 @@
+:toc: macro
+
+= RFC 5: Emptying and closing wallets
+
+:icons: font
+:numbered:
+toc::[]
+
+== Overview
+
+If a wallet still containing BTC is to be closed,
+the remaining BTC must be transferred to a new wallet if possible.
+
+This applies in the following situations:
+
+- a wallet has fallen below a specified minimum BTC amount
+- a wallet has failed a heartbeat due to losing too many operators
+- a wallet has reached a specified maximum lifetime
+
+When one of these _closure criteria_ is met,
+the wallet changes state to `Emptying`
+and must produce a transaction moving its remaining BTC to a newer wallet.
+
+=== Emptying wallets
+
+When a wallet has began `Emptying`
+the following things change:
+
+- If the wallet was previously a valid deposit wallet,
+it stops being a valid deposit wallet.
+- No redemptions can be made from the wallet.
+- The wallet has 24h to sweep any remaining unswept deposits
+and 48h to transfer its remaining BTC to a newer wallet.
+- The time the wallet began emptying may be recorded,
+or it may be calculated from the time the wallet reached its maximum age
+or reached a timeout in some operation.
+
+==== Sweeping and rejecting deposits
+
+In the scenario where a previously valid deposit wallet begins emptying,
+it stops receiving deposits.
+Any deposits erroneously made to an emptying wallet after it began emptying
+will be rejected and not add to the depositor's BTC balance,
+and must be allowed to time out.
+
+Any deposits made before the wallet began emptying,
+if the wallet was a valid deposit wallet when the deposit was made,
+must be swept within 24h of the wallet beginning to empty.
+This overrides any other deposit sweeping timeouts
+that would give the wallet more time to complete the sweep.
+If the deposit had an earlier timeout,
+the original timeout is unchanged.
+The submitter/s of transaction/s proving the sweeps are rewarded as usual.
+
+Failure to sweep a remaining deposit within the allowed 24h
+is treated like any other failure to sweep.
```

Comment:
We do nothing because depositors can use the refund option from their BTC script, correct?

Reply:
We slash the operators, I believe. Normally they have more time before slashing starts happing, but in this case we're saying that slashing happens sooner.

Reply:
Do we have slashing rules listed somewhere? I think it could be a good candidate for another RFC.

Reply:
Slashing is upcoming